### PR TITLE
Feat/connect

### PR DESCRIPTION
**Pull request by Blaxwo**
## Describe your changes

- I connected backend and frontend for sign-in
-  I connected backend and frontend for restore password page
- I added implementation of taosts
- I fixed files structure (moved into pages their forms)

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-40?atlOrigin=eyJpIjoiMzE5OWMwY2I3Y2JjNGE4NTgyOTRmY2FjYjVjODY4ZjIiLCJwIjoiaiJ9)
- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-43?atlOrigin=eyJpIjoiZmVhZDRlMzFlYjIxNGNiMDg5ZmNlZWQwZjc4YzRiYTciLCJwIjoiaiJ9)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
зображення

зображення

### CODE REVIEW

File: src/components/NewPasswordForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import { useForm, Controller, SubmitHandler } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { Box, Typography } from '@mui/material';
+import PasswordInput from 'src/components/shared/PasswordInput';
```
Comment (lyudad) [inline src/components/NewPasswordForm/index.tsx:8]: empty line above

File: src/components/NewPasswordForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  InputStyleVariants,
+} from 'src/components/shared/StyledInput/types';
+import StyledButton from 'src/components/shared/StyledButton';
+import {
+  PaddingVariants,
+  StyleVariants,
+} from 'src/components/shared/StyledButton/types';
+import LabelText from 'src/components/shared/LabelText';
+import TitleInputWrapper from 'src/components/shared/TitleInputWrapper';
+import { styled } from '@mui/system';
```
Comment (lyudad) [inline src/components/NewPasswordForm/index.tsx]: it in wrong place

File: src/components/SignInForm/index.tsx
Comment (lyudad) [inline src/components/SignInForm/index.tsx:1]: I assume sign in page have only sign in form, why this component so far from the page?

File: src/components/RestorePasswordForm/index.tsx
Comment (lyudad) [inline src/components/RestorePasswordForm/index.tsx:1]: I assume sign in page have only sign in form, why this component so far from the page?

File: src/components/NewPasswordForm/index.tsx
Comment (lyudad) [inline src/components/NewPasswordForm/index.tsx:1]: I assume sign in page have only sign in form, why this component so far from the page?

File: src/components/SignInForm/index.tsx
Comment (Blaxwo) [reply src/components/SignInForm/index.tsx:1]: Understood, I will replace  sign in form in Sign In Page and will do the same thing with other forms

**Review by lyudad — APPROVED** [review]

**Pull request merged**